clang/include/clang/AST/Type.h

```diff
@@ -7035,17 +7035,16 @@ class DependentNameType : public TypeWithKeyword, public llvm::FoldingSetNode {
       : TypeWithKeyword(Keyword, DependentName, CanonType,
                         TypeDependence::DependentInstantiation |
                             toTypeDependence(NNS->getDependence())),
-        NNS(NNS), Name(Name) {}
+        NNS(NNS), Name(Name) {
+    assert(Name);
```

Comment:
FWIW NNS must also be non-null, and additionally it must also be dependent.

Reply:
> FWIW NNS must also be non-null

Thanks, updated the patch to assert this.

> and additionally it must also be dependent.

According to this comment we can't assert this one unconditionally:

https://github.com/llvm/llvm-project/blob/1de3dc7d23dd6b856efad3a3a04f2396328726d7/clang/include/clang/AST/Type.h#L7026-L7028